x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/asset_details_flyout.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { Dispatch, SetStateAction } from 'react';
+import AssetDetails from '../../../../../components/asset_details/asset_details';
+import { useSourceContext } from '../../../../../containers/metrics_source';
+import { useWaffleTimeContext } from '../../hooks/use_waffle_time';
+import { orderedFlyoutTabs } from './flyout_tabs';
+
+interface Props {
+  assetName: string;
+  closeFlyout: () => void;
+  setIsAlertFlyoutVisible: Dispatch<SetStateAction<boolean>>;
```

Comment:
how about if the event here was named like `onCreateRuleClick`. It might be easier to understand from the caller that when this event is handled, it's actually about the alert rule button click.

It would also give it more flexibility - for instance, if something else other than setting the flyout visibility should happen.

e.g:

```tsx
const handleCreateRuleClick = () => {
   setIsAlertFlyoutVisible(true)
}

<AssetDetailsFlyout
  ...
  onCreateRuleClick={handleCreateRuleClick}
/>
```